frontend/src/app/tasks/page.tsx

```diff
@@ -14,8 +15,14 @@ const getPageData = async () => {
   ])
 }
 
+const authorizedRoles = [ "staff", "manager", "admin", "project manager", "human resources"]
```

Comment:
In order not to have this hardcoded, how would we solve it? I vaguely recall discussing this with you, would it be creating a map of routes and roles and storing in the db?

Reply:
I am still thinking about this part. I am leaning towards something like what you mention, but I have also seen applications put something like this in the middleware - with hardcoded strings or an enum. I was initially worried about bloating the middleware, but it could be worth exploring doing the check there. 

I have a couple of other changes I think I am going to make before pulling this out of draft status, so I can experiment a little.

Reply:
The problem is that we need this check in the backend too so in any case we need them stored somewhere no only in the middleware.

I'll try to do some research too for solutions that don't over complicated.